Hw 1

olegastra-git opened the pull request:
домашняя работа по phiton задачи 26 и 28

olegastra-git committed:
домашняя работа задача 26

olegastra-git committed:
Домашняя работа задача 28

olegastra-git committed:
задача 30 для самопроверки

olegastra-git committed:
задача 32 для самопроверки

samodurOFF commented on HW_1/26.py:
```diff
@@ -0,0 +1,13 @@
+
+a = int(input(" vvedite chislo a: - "))
+b = int(input(" vvedite chislo b: - "))
+def power(a,b):
```
```suggestion
def power(a, b):
```

samodurOFF commented on HW_1/26.py:
```diff
@@ -0,0 +1,13 @@
+    if b == 0:
+        return 1
+    else:
+        return a* power(a, b-1)
```
```suggestion
        return a * power(a, b - 1)
```

samodurOFF commented on HW_1/26.py:
```diff
@@ -0,0 +1,13 @@
+        return 1
+    else:
+        return a* power(a, b-1)
+result=power(a,b)
```
```suggestion
result = power(a, b)
```

samodurOFF commented on HW_1/28.py:
```diff
@@ -0,0 +1,7 @@
+a = int(input(" vvedite chislo a: - "))
+b = int(input(" vvedite chislo b: - "))
+def summa(a, b):
+    return summa(a + 1, b - 1) if b > 0 else
+   print sum(a, b)
```
Вот ту не понял. Тут съехали отступы и нет кода после `else`

samodurOFF commented on HW_1/30.py:
Это задача не из 5 ДЗ

samodurOFF commented on HW_1/32.py:
И эта тоже

olegastra-git committed:
Задача 34:  Винни-Пух попросил Вас посмотреть, есть ли в его стихах ритм. Поскольку разобраться в его кричалках не настолько просто, насколько легко он их придумывает, Вам стоит написать программу. Винни-Пух считает, что ритм есть, если число слогов (т.е. число гласных букв) в каждой фразе стихотворения одинаковое. Фраза может состоять из одного слова, если во фразе несколько слов, то они разделяются дефисами. Фразы отделяются друг от друга пробелами. Стихотворение  Винни-Пух вбивает в программу с клавиатуры. В ответе напишите “Парам пам-пам”, если с ритмом все в порядке и “Пам парам”, если с ритмом все не в порядке

olegastra-git committed:
Задача 34:  Винни-Пух попросил Вас посмотреть, есть ли в его стихах ритм. Поскольку разобраться в его кричалках не настолько просто, насколько легко он их придумывает, Вам стоит написать программу. Винни-Пух считает, что ритм есть, если число слогов (т.е. число гласных букв) в каждой фразе стихотворения одинаковое. Фраза может состоять из одного слова, если во фразе несколько слов, то они разделяются дефисами. Фразы отделяются друг от друга пробелами. Стихотворение  Винни-Пух вбивает в программу с клавиатуры. В ответе напишите “Парам пам-пам”, если с ритмом все в порядке и “Пам парам”, если с ритмом все не в порядке
Задача 36: Напишите функцию print_operation_table(operation, num_rows=6, num_columns=6), которая принимает в качестве аргумента функцию, вычисляющую элемент по номеру строки и столбца. Аргументы num_rows и num_columns указывают число строк и столбцов таблицы, которые должны быть распечатаны. Нумерация строк и столбцов идет с единицы (подумайте, почему не с нуля). Примечание: бинарной операцией называется любая операция, у которой ровно два аргумента, как, например, у операции умножения.

samodurOFF commented on HW_2/34.py:
```diff
@@ -0,0 +1,9 @@
+def f(words):
+    return sum(1 for i in words if i in 'аеёиоуыэюя')
```
`    return sum(i in 'аеёиоуыэюя' for i in words)`

samodurOFF commented on HW_2/34.py:
Задание решено верно. Однако это домашнее задание выдается после темы о функциях высшего порядка. Следовательно, здесь рекомендуется использовать именно этот инструмент. Попробуйте решить именно так.

samodurOFF commented on HW_2/36.py:
Отлично. Но попробуйте решить, используя один цикл `for`.

samodurOFF reviewed (commented):
Олег, не сливает изменения до моей проверки, иначе я не могу вам предлагать изменение кода.
